proposals/0006-nio-apns-api.md

```diff
@@ -0,0 +1,238 @@
+# NIO-based Apple Push Notification Service
+
+* Proposal: SSWG-0006
+* Authors: [Kyle Browning](https://github.com/kylebrowning)
+* Sponsors: TBD
+* Review Manager: TBD
+* Status: **Implemented**
+* Pitch: [Server/Pitches/APNS Implementation](https://forums.swift.org/t/apple-push-notification-service-implementation-pitch/20193)
+* Implementation: [kylebrowning/swift-nio-http2-apns](https://github.com/kylebrowning/swift-nio-http2-apns)
+
+## Package Description
+Apple push notification service implementation built on Swift NIO.
+
+|  |  |
+|--|--|
+| **Package name** | `nio-apns` |
+| **Proposed Maturity Level** | [Sandbox](https://github.com/swift-server/sswg/blob/master/process/incubation.md#process-diagram) |
+| **License** | [Apache 2](https://www.apache.org/licenses/LICENSE-2.0.html) |
+| **Dependencies** | [SwiftNIO](https://github.com/apple/swift-nio) 2.x, [SwiftNIOSSL](https://github.com/apple/swift-nio-ssl.git) 2.x, [SwiftNIOHTTP2](https://github.com/apple/swift-nio-http2.git) 1.x|
+
+## Introduction
+
+`NIOAPNS` is a module thats gives server side swift applications the ability to use the [Apple push notification service.](https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server)
+
+
+## Motivation
+
+APNS is used to push billions of pushes a day, (7 billion per day in 2012). Many of us using Swift on the backend are using it to power our iOS applications. Having a community supported APNS implementation would go a long way to making it the fastest, free-ist, and simplest solution that exists.
+
+All of the currently maintained libraries either have framework specific dependencies, are not built with NIO, or do not provide enough extensibility while providing "out of the box" capabilities.
+
+### Existing Solutions
+
+- [NIO-APNS](https://github.com/moritzsternemann/nio-apns)
+- [PerfectNotifications](https://github.com/PerfectlySoft/Perfect-Notifications)
+- [VaporNotifications](https://github.com/hjuraev/VaporNotifications)
+
+## Proposed Solution
+
+`NIOApns` provides the essential types for interacting with APNS Server (both production and sandbox).
```

Comment:
Swift naming guidelines would make this `NIOAPNS`

Reply:
@kylebrowning okay if I comment here?

Reply:
Of course.